src/packagedcode/phpcomposer.py

```diff
@@ -336,3 +354,56 @@ def parse_person(persons):
                 url and url.strip('() '))
     else:
         raise ValueError('Incorrect PHP composer persons: %(persons)r' % locals())
+
+
+def build_package_from_lock(package_data):
+    """
+    Yield a composer Package object from a package data mapping that originated
+    from a composer.lock file
+    """
+    packages = package_data.get('packages', []) + package_data.get('packages-dev', [])
+    for package in packages:
+        dependencies = []
+        for dep, req in package.get('require', {}).items():
+            dependencies.append(
+                models.DependentPackage(
+                    purl=PackageURL(type='composer', name=dep).to_string(),
```

Comment:
The dependencies in a lock file would be special because "is_resolved" would be True. that is all the packages in the lock file are eventually the deps that are have been resolved?
(with care not to be mixing up the deps in the package reord of a lock file entry and a lock file entry which is itself a dependent Package of the root Package, e.g. the one in the composer.json)